### PR TITLE
Updating facebook-swift-sdk to Swift5

### PR DESCRIPTION
**Pull request by cdeust**
## Checklist

- [X] I've read the [Contributing Guidelines](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md)
- [X] I've completed the [Contributor License Agreement](https://developers.facebook.com/opensource/cla)
- [X] I've ensured that all existing tests pass and added tests (when/where necessary)
- [  ] I've ensured that my code lints properly: (`swiftlint && swiftlint autocorrect --format`)
- [  ] I've updated the documentation (when/where necessary) and [Changelog](CHANGELOG.md) (when/where necessary)
- [X] I've added the proper label to this pull request (e.g. `bug` for bug fixes)

## Pull Request Details

I've updated the project settings to comply with Swift 5 Compiler and made the changes needed inside the projects to have all frameworks building with Swift 5 Compiler.
I've updated properties said to be deprecated with __deprecated when they are deprecated properties so it complies with Apple naming.

### CODE REVIEW

File: Sources/Core/Common/SDKSettings.swift [hunk 1]
```diff
@@ -148,7 +148,7 @@ public enum SDKSettings {
       }
 
       #if swift(>=4.1)
-      let behaviors: [SDKLoggingBehavior] = FBSDKSettings.loggingBehavior().compactMap(createBehavior)
+      let behaviors: [SDKLoggingBehavior] = FBSDKSettings.loggingBehaviors.compactMap(createBehavior)
```
Comment (codytwinton) [inline Sources/Core/Common/SDKSettings.swift]: What is the purpose of this change? I believe that loggingBehavior remains a function in `v4.42.0`.

File: Sources/Share/Views/LikeControl.swift [hunk 2]
```diff
@@ -64,10 +64,10 @@ public class LikeControl: UIView {
   /// The foreground color to use for the content of the control.
   public var foregroundColor: UIColor {
     get {
-      return sdkLikeControl.foregroundColor
+      return sdkLikeControl.foregroundColor__deprecated
```
Comment (codytwinton) [inline Sources/Share/Views/LikeControl.swift]: Can you explain this change?

**Comment by codytwinton** [comment]
@cdeust - Any thoughts on this review?

File: Sources/Share/Views/LikeControl.swift (hunk 2 above)
Comment (cdeust) [reply Sources/Share/Views/LikeControl.swift]: According to Apple you need to add the __deprecated flag to properties that are deprecated so developer are aware they are using deprecated methods and properties.

File: Sources/Core/Common/SDKSettings.swift (hunk 1 above)
Comment (cdeust) [reply Sources/Core/Common/SDKSettings.swift]: In Swift5 FBSDKSettings.loggingBehavior().compactMap(createBehavior) as been renamed FBSDKSettings.loggingBehaviors.compactMap(createBehavior)

File: Sources/Share/Views/LikeControl.swift (hunk 2 above)
Comment (joesus) [reply Sources/Share/Views/LikeControl.swift]: Trying to catch up here. From a little experimenting it seems that adding the __deprecated flag to the objc header will  mark the swift method as deprecated. Ex:

```swift
/**
  The foreground color to use for the content of the receiver.
 */
@property (nonatomic, strong) UIColor *foregroundColor __deprecated;
```
will generate this Swift header:
```swift
/**
  The foreground color to use for the content of the receiver.
 */
@available(*, deprecated)
open var foregroundColor: UIColor!
```

I wasn't able to find anything about the __deprecated flag that would imply that it works in reverse. Could you point me to what you read so I can get caught up? All I was able to find was this post from a while back. https://forums.developer.apple.com/thread/5244 Also, Thanks for helping out with this!

Comment (robtimp) [reply Sources/Share/Views/LikeControl.swift]: Looks like this change is causing the build to fail for `FacebookShare`. There's also new warnings for missing nullability annotations that are treated as errors.

**Comment by Gujci** [comment]
@cdeust I think you also have to modify `osx_image: xcode10` to `osx_image: xcode10.2` in the travis file in order to pass builds. Travis updated their xcode image around this weekend (last weekend of march), so it was after you have submitted this PR.

**Comment by cdeust** [comment]
Updated the pull request and tested out that it works as it is now with Swift5 Compiler and XCode 10.2

**Comment by codytwinton** [comment]
@cdeust - Thank you so much for submitting this PR. This brought to light a few things we were missing and I really appreciate it.

Version `0.6.0` was just released for Carthage and Cocoapods with support for Xcode 10.2 and Swift 5.0.
